linux/deploy_vm/rebuild_unattend_install_iso.yml

```diff
@@ -339,3 +339,142 @@
+        - "sed -i '#./boot/grub/grub.cfg#d' md5sum.txt"
+        - "echo \"`md5sum grub.cfg | awk '{print $1}'` ./boot/grub/grub.cfg\" >>md5sum.txt"
+        - "sed -i '#./isolinux/install_en.cfg#d' md5sum.txt"
+        - "echo \"`md5sum install_en.cfg | awk '{print $1}'` ./isolinux/install_en.cfg\" >>md5sum.txt"
+        - "sed -i '#./isolinux/install.cfg#d' md5sum.txt"
+        - "echo \"`md5sum install.cfg | awk '{print $1}'` ./isolinux/install.cfg\" >>md5sum.txt"
+      args:
+        chdir: "{{ src_iso_file_dir }}"
+      register: update_initrd_output
+
+    - name: "Print command output for updating initrd"
+      ansible.builtin.debug: var=update_initrd_output
+    
+    - name: "Get file to get code name"
+      ansible.builtin.shell: "cat {{ src_iso_file_dir }}/md5sum.txt | grep -E 'dists\\/[a-zA-Z]+\\/Release$' | cut -d. -f 2"
+      register: release_file_path
+    
+    - name: "Print release file path"
+      ansible.builtin.debug: var=release_file_path
+    
+    - name: "Extract specific release file inside ISO"
+      community.general.iso_extract:
+        image: "{{ src_iso_file_path }}"
+        dest: "{{ src_iso_file_dir }}"
+        files:
+          - "{{ release_file_path.stdout.split('/', 1)[-1] }}"
+    
+    - name: "Get codename of the ISO distribution"
+      ansible.builtin.shell: "cat {{ src_iso_file_dir }}/Release | grep Codename | cut -d: -f 2"
+      register: codename_output
+    
+    - name: "Set facts of the codename"
+      ansible.builtin.set_fact:
+        os_codename: "{{ codename_output.stdout.strip() }}"
+
+    - name: "Print codename"
+      ansible.builtin.debug: var=os_codename
+
+    - name: "Replace the codename in file preseed.cfg"
+      ansible.builtin.replace:
+        path: "{{ new_unattend_install_conf }}"
+        regexp: "yirmibir"
+        replace: "{{ os_codename }}"
+    
+    - name: "Print the content of unattend install config file after replace codename"
+      ansible.builtin.debug:
+        msg: "{{ lookup('file', new_unattend_install_conf) | split('\n') }}"
+
+    - name: "Customize the ISO"
+      community.general.iso_customize:
+        src_iso: "{{ src_iso_file_path }}"
+        dest_iso: "{{ rebuilt_unattend_iso_path }}"
+        add_files:
+          - src_file: "{{ src_iso_file_dir }}/grub.cfg"
+            dest_file: "boot/grub/grub.cfg"
+          - src_file: "{{ src_iso_file_dir }}/isolinux.cfg"
+            dest_file: "isolinux/isolinux.cfg"
+          - src_file: "{{ src_iso_file_dir }}/install_en.cfg"
+            dest_file: "isolinux/install_en.cfg"
+          - src_file: "{{ src_iso_file_dir }}/install.cfg"
+            dest_file: "isolinux/install.cfg"
+          - src_file: "{{ src_iso_file_dir }}/md5sum.txt"
+            dest_file: "md5sum.txt"
+          - src_file: "{{ new_unattend_install_conf }}"
+            dest_file: "{{ unattend_install_file_name }}"
+  when: unattend_install_conf is match('Pardus')
```

Comment:
This file is getting longer. I think we can consider splitting it from line 31 to the end for Ubuntu/Photon/Debian/FreeBSD/Pardus. For example, for you added block of `Rebuild ISO for Pardus`, you can put it into a new file like:
```
rebuild_pardus_unattend_install_iso.yml
```
And the others could be:
```
rebuild_ubuntu_unattend_install_iso.yml
rebuild_photon_unattend_install_iso.yml
rebuild_debian_unattend_install_iso.yml
rebuild_freebsd_unattend_install_iso.yml
```

Before including these tasks files, you can extract the OS name from unattend_install_conf and set it as a fact like:
```
unattend_install_os_type: "{{ unattend_install_conf.split('/')[0]  | lower }}"
```
Then you can include those child task files like:
```
 include_tasks: "rebuild_{{ unattend_install_os_type }}_unattend_install_iso.yml"
```

Reply:
Change it.
The testing is on-going for debian, photon, ubuntu, pardus.

Best regards
Yuhua Zou